### ISSUE TITLE
Fatal assert failed in pipes.cpp

### ISSUE THREAD

**Issue by oskarkv** [ValveSoftware/Dota-2#1347]
Got this at about Tue Nov 14 16:30:00 CET 2017 while in game in DOTA 2.

```
~ » ../common/pipes.cpp (766) : CClientPipe::BWriteAndReadResult: BWaitResult failed, disconnected
../common/pipes.cpp (766) : Fatal assert failed: ../common/pipes.cpp, line 766.  Application exiting.

_ExitOnFatalAssert

```

**Comment by gdrewb-valve** [comment]
What else happened on your system at that time?  Did Dota exit (it should have)?  Did the Steam client exit (the assert appears to the the Steam client exiting, causing Dota to exit)?

**Comment by oskarkv** [comment]
Yes, they both exited (Dota and Steam). And I'm on Linux Mint by the way.

**Comment by gdrewb-valve** [comment]
It seems like this might be a Steam issue so I've assigned this to a Steam team dev.

**Comment by Th3BiSh0p** [comment]
same issue with another game. Anything new since november ?

**Comment by AiyionPrime** [comment]
Have the same issue while running Borderlands 2 for over an hour.

**Comment by saucecode** [comment]
This problem still happens! I'm able to induce it while running any Steam game by physically disconnecting and reconnecting my computer's internet. Because Steam crashes, it also kills the game I'm in. Very annoying!
